docs/internals/frontends.rst

```diff
@@ -0,0 +1,159 @@
+.. include:: ../global.rst.inc
+.. highlight:: none
+
+.. _json_output:
+
+All about JSON: How to develop frontends
+========================================
+
+Borg does not have a public API on the Python level. That does not keep you from writing :code:`import borg`,
+but does mean that there are no release-to-release guarantees on what you might find in that package, not
+even for point releases (1.1.x), and there is no documentation beyond the code and the internals documents.
+
+Borg does on the other hand provide an API on a command-line level. In other words, a frontend should to
+(for example) create a backup archive just invoke :ref:`borg_create`.
+
+Logging
+-------
+
+Especially for graphical frontends it is important to be able to convey and reformat progress information
+in meaningful ways. The ``--log-json`` option turns the stderr stream of Borg into a stream of JSON lines,
+where each line is a JSON object. The *type* key of the object determines its other contents.
```

Comment:
somehow a (non) stability assertion is missing here - is it expected to change?

if so, would a json output of `borg --version` be useful?